WeightProducer/python/MCSampleValues.py

```diff
@@ -63,493 +63,1039 @@ class MCSampleValuesHelper():
+            ),
+            "kFactor" : kFactorValues(
+                kFactor_2016=1.224, kFactorSource_2016="XSDB NNLO/LO=61526.7/50260",
+                kFactor_2017=1.162, kFactorSource_2017="XSDB NNLO/LO=61526.7/52940",
+                kFactor_2018=1.164, kFactorSource_2018="XSDB NNLO/LO=61526.7/52850",
+            ),
         },
         "WJetsToLNu_HT-1200To2500" : {
-            "CrossSection" : XSValues(XS_13TeV=1.608),
-            "kFactor" : kFactorValues(kFactor_13TeV=1.21),
+            "CrossSection" : XSValues(
+                XS_2016=1.329, XSSource_2016="XSDB (LO)",
+                XS_2017=1.074, XSSource_2017="XSDB (LO)",
+                XS_2018=1.085, XSSource_2018="GenXSecAnalyzer",
+            ),
+            "kFactor" : kFactorValues(
+                kFactor_2016=1.224, kFactorSource_2016="XSDB NNLO/LO=61526.7/50260",
+                kFactor_2017=1.162, kFactorSource_2017="XSDB NNLO/LO=61526.7/52940",
+                kFactor_2018=1.164, kFactorSource_2018="XSDB NNLO/LO=61526.7/52850",
+            ),
         },
         "WJetsToLNu_HT-2500ToInf" : {
-            "CrossSection" : XSValues(XS_13TeV=0.03891),
-            "kFactor" : kFactorValues(kFactor_13TeV=1.21),
+            "CrossSection" : XSValues(
+                XS_2016=0.03209, XSSource_2016="XSDB (LO)",
+                XS_2017=0.008001, XSSource_2017="XSDB (LO)",
+                XS_2018=0.008060, XSSource_2018="GenXSecAnalyzer",
+            ),
+            "kFactor" : kFactorValues(
+                kFactor_2016=1.224, kFactorSource_2016="XSDB NNLO/LO=61526.7/50260",
+                kFactor_2017=1.162, kFactorSource_2017="XSDB NNLO/LO=61526.7/52940",
+                kFactor_2018=1.164, kFactorSource_2018="XSDB NNLO/LO=61526.7/52850",
+            ),
         },
         "WJetsToLNu" : {
-            "CrossSection" : XSValues(XS_13TeV=61334.9),
-        },
-        "WJetsToQQ_HT400to600_qc19_3j" : {
-            "CrossSection" : XSValues(XS_13TeV=316.4),
-        },
-        "WJetsToQQ_HT600to800_qc19_3j" : {
-            "CrossSection" : XSValues(XS_13TeV=68.54),
-        },
-        "WJetsToQQ_HT-800toInf_qc19_3j" : {
-            "CrossSection" : XSValues(XS_13TeV=34.68),
-        },
-        "WJetsToQQ_HT-600ToInf" : {
-            "CrossSection" : XSValues(XS_13TeV=95.14),
-            "kFactor" : kFactorValues(kFactor_13TeV=1.21),
+            "CrossSection" : XSValues(
+                XS_13TeV=61526.7, XSSource_13TeV="https://twiki.cern.ch/twiki/bin/viewauth/CMS/SummaryTable1G25ns#W_jets NNLO (60430.0 @ NLO)",
+                XS_2016=50260.0, XSSource_2016="XSDB (LO)",
+                XS_2017=52940.0, XSSource_2017="XSDB (LO)",
+                XS_2018=52850.0, XSSource_2018="XSDB (LO)",
+            ),
+        },
+        "QCD_FlatPt_15_3000" : {
+            "CrossSection" : XSValues(
+                XS_2017=1985000000.0, XSSource_2017="XSDB (LO)",
+                XS_2018=1370000000.0, XSSource_2018="XSDB (LO)",
```

Comment:
also surprised at this change 2017->2018

Reply:
This has more to do with the change in how the samples are flattened from year to year. I'm not so surprised by this.